cdk/lib/prism.ts

```diff
@@ -48,27 +49,33 @@ export class PrismStack extends GuStack {
     });
 
     const distBucket: string = this.getParam(GuDistributionBucketParameter.parameterName).valueAsString;
-    const s3Key = [distBucket, this.stack, this.stage, this.app, `${this.app}.deb`].join("/");
 
-    // TODO move to UserData.forLinux
-    const userData = `#!/bin/bash -ev
-
-      aws --region ${this.region} s3 cp s3://${s3Key} /tmp/
-      dpkg -i /tmp/prism.deb`;
+    const userData = new GuUserData(this, {
+      distributable: {
+        bucketName: distBucket,
+        fileName: "prism.deb",
+        executionStatement: `dpkg -i /${this.app}/prism.deb`,
+      },
+    });
 
     const asg = new GuAutoScalingGroup(this, "AutoscalingGroup", {
       overrideId: true,
       vpc,
       vpcSubnets: { subnets },
       role: role,
-      userData: userData,
-      minCapacity: 2,
-      maxCapacity: 4,
-      desiredCapacity: 2,
+      userData: userData.userData,
+      stageDependentProps: {
+        [Stage.CODE]: {
+          minimumInstances: 2,
+        },
+        [Stage.PROD]: {
+          minimumInstances: 2,
+        },
+      },
       healthCheck: HealthCheck.elb({
         grace: Duration.seconds(500),
       }),
-      securityGroup: appServerSecurityGroup,
+      additionalSecurityGroups: [appServerSecurityGroup],
```

Comment:
We need to keep this for now (we cannot replace a security group in a single CloudFormation update). It's more permissive than the new default security group (see https://github.com/guardian/cdk/pull/268), but I wonder if we can remove it entirely in the future?